CONTRIBUTING.md

```diff
@@ -188,7 +188,49 @@ with one another are rolled up.
 Speaking of tests, Rust has a comprehensive test suite. More information about
 it can be found [here][rctd].
 
-### External Dependencies
+### External Dependencies (subtree)
+
+As a developer to this repository, you don't have to treat the following external projects
+differently from other crates that are directly in this repo:
+
+* none so far, see https://github.com/rust-lang/rust/issues/70651 for more info
+
+They are just regular files and directories. This is in contrast to `submodule` dependencies
+(see below for those).
+
+#### Synchronizing a subtree
+
+There are two synchronization directions: `subtree push` and `subtree pull`.
+
+A `git subtree push -P src/tools/clippy`
+takes all the changes that
+happened to the copy in this repo and creates commits on the remote repo that match the local
+changes (so every local commit that touched the subtree causes a commit on the remote repo).
+
+A `git subtree pull -P src/tools/clippy` takes all changes since the last `subtree pull` from the clippy
+repo and creates a single commit in the rustc repo with all the changes.
```

Comment:
Two things here:

1. From the [git documentation](), the syntax for these commands is
    ```
    'git subtree' pull  -P <prefix> <repository> <ref>
    'git subtree' push  -P <prefix> <repository> <ref>
    ```
    I think it is also important to note for `push`, that this shouldn't be done directly to the Clippy master branch, but to a new branch + PR, so it will get checked by the Clippy CI (formatting, integration tests, ...)
2. "creates a single commit in the rustc repo with all the changes." is not true for `git subtree pull`, is it?